mesonbuild/dependencies/base.py

```diff
@@ -1411,7 +1411,10 @@ def _detect_dep(self, name: str, modules: T.List[T.Tuple[str, bool]], components
                                       'Valid targets are:\n{}'.format(name, list(self.traceparser.targets.keys())))
 
         # Set dependencies with CMake targets
-        reg_is_lib = re.compile(r'^(-l[a-zA-Z0-9_]+|-pthread)$')
+        if mesonlib.is_windows():
+            reg_is_lib = re.compile(r'(^-delayload:[a-zA-Z0-9_\.]+|[a-zA-Z0-9_]+\.(lib|dll))$')
```

Comment:
Could you combine this into one big regex? `is_windows()` Is also true on MSYS and doesn't necessarily imply the use of Microsofts compiler.

Reply:
So just have a single regex for all systems, matching all of -lfoo, -delayload:foo, foo.lib and foo.dll?

Reply:
Yes.